packages/core-web/src/index.js

```diff
@@ -83,7 +83,20 @@ function detectAndApplyFixedHeaderStyles() {
     });
     resizeObserver.observe(headerSelector[0]);
   };
+
+  let lastOffset = 0;
+  const toggleHeaderOnScroll = () => {
+    const currentOffset = window.pageYOffset;
+    if (currentOffset > lastOffset) {
+      headerSelector.addClass('hide-header');
+    } else {
+      headerSelector.removeClass('hide-header');
+    }
+    lastOffset = currentOffset;
+  };
+
   addResizeHeaderListener();
+  setTimeout(() => window.addEventListener('scroll', toggleHeaderOnScroll), 500);
```

Comment:
Can I check why do we need this setTimeout here? :O

Reply:
This is for rare situations where the size of the navbar can change when the page is refreshed. For example, when a `<box dismissable>` is placed in the header. This will cause the navbar to change size while rendering resulting in an unwanted scroll event to fire before the page is fully loaded. Although not the best way to counter this, a delay here would prevent the unwanted scroll event to be picked up. I believe the timeout would not be necessary anymore after #1406 is implemented.

Reply:
https://user-images.githubusercontent.com/47470981/107607087-2a33aa80-6c73-11eb-881a-33ff299bc27a.mp4

Reply:
Thanks for the explanation! I'm not sure if `setTimeout` would be appropriate in the case where the user has slow loading speed (e.g. page takes longer than the stipulated 500ms to load). I think the solution here won't work in this scenario. However, I'm also unsure if there is a better way to solve this issue :-/ 

I tried looking through the list of event hooks that we can possibly use to detect when the page is fully-rendered but I don't think I found any.

https://developer.mozilla.org/en-US/docs/Web/Events

I also tried a few other hacky ways to get a solution to work but no luck so far. 

But a surefire way to solve this problem at some expense can be `window.history.scrollRestoration = 'manual';`, where we always start off at the top whenever a page is reloaded, instead of where the user was at previously. I'm not sure if this is the way we want to go forward but I think it's a potential solution.

Reply:
Thanks for investigating this @wxwxwxwx9 👍. Since this issue is only apparent on desktop version of the navbar, if we follow @raysonkoh suggestion of only hiding the header on mobile, then the unwanted scroll event will not be captured. I've tested the mobile version and it works fine as the box is contained within the dropdown so a reload will not trigger a scroll.

Reply:
Alright! That would work as well. :-)